src/main/java/org/apache/commons/lang3/LocaleUtils.java

```diff
@@ -248,7 +248,9 @@ private static Locale parseLocale(final String str) {
             return new Locale(str);
         }
 
-        final String[] segments = str.split("_", -1);
+        final String[] segments = str.contains("-")
+            ? str.split("-", -1)
```

Comment:
Blame my weak eyes - could we introduce constants here, e.g. "UNDERSCORE"?

Reply:
Done in 11a6d1d52.